process/block/gasConsumption.go

```diff
@@ -292,6 +308,13 @@ func (gc *gasConsumption) addPendingIncomingMiniBlocks() ([]data.MiniBlockHeader
 
 		addedMiniBlocks = append(addedMiniBlocks, mb)
 		lastIndexAdded = i
+
+		gasConsumedByIncomingAfter := gc.totalGasConsumed[incoming]
+		gasConsumedByMiniBlock := gasConsumedByIncomingAfter - gasConsumedByIncomingBefore
+		if gasConsumedByMiniBlock < gc.totalGasConsumed[pending] {
```

Comment:
for the last pending miniblock I think this condition would be equal.

this relates to the comment below

// should never be false

but for the last one, it will be false.

maybe the conditon should be  `<=` gc.totalGasConsumed[pending]

Reply:
updated